### PR TITLE
Add debugging flags to enhance the timeline arguments for Build, Layout, and Paint

### LINKED ISSUE #101535: Do not send debugTimelineArguments when `debugProfileBuildsEnabled == true`
While debugging https://github.com/flutter/flutter/issues/101379, I noticed that with Track Widget Builds enabled (`debugProfileBuildsEnabled`), a large amount of data is getting sent in the trace event args for the 'BUILD' timeline event: https://github.com/flutter/flutter/blob/35b18ba2e902dac1e16f557fa385069f1f5e1069/packages/flutter/lib/src/widgets/framework.dart#L2568-L2575

These additional debugging args were added in https://github.com/flutter/flutter/pull/93086. These events make the trace very heavy and will make the performance impact of Track Widget Builds even worse than it already is. We should create another flag to hide these args behind instead of reusing `debugProfileBuildsEnabled`. 

This may apply to other `debugTimelineArguments` added in https://github.com/flutter/flutter/pull/93086/files#diff-134bc36bd7c045a92ffff29f7efdaa7d375384911475a40343814c76c87b17cb as well. 

CC @Hixie @goderbauer @jacob314

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Fixes https://github.com/flutter/flutter/issues/101535. The extra arguments were previously hidden behind the `debugProfileBuildsEnabled`, `debugProfileLayoutsEnabled`, and `debugProfilePaintsEnabled` flags. Now the additional arguments. will be hidden behind their own respective debugging flags -- the additional event arguments make the timeline trace very heavy. `debugProfileBuildsEnabled`, `debugProfileLayoutsEnabled`, and `debugProfilePaintsEnabled` are useful without the extra arguments, so we want to make these features useable without severely impacting the app's performance due to tracing overhead.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/debug.dart [hunk 1]
```diff
@@ -132,6 +132,17 @@ bool debugProfileBuildsEnabled = false;
 ///    for every widget and has a higher overhead cost.
 bool debugProfileBuildsEnabledUserWidgets = false;
 
+/// Adds debugging information to the args of [Timeline] events related to
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/debug.dart:135]: nit: args -> arguments? Or we could just completely omit the "the args of".

File: packages/flutter/lib/src/widgets/debug.dart [hunk 2]
```diff
@@ -132,6 +132,17 @@ bool debugProfileBuildsEnabled = false;
...
+/// [Widget] builds.
+///
+/// This flag will only add [Timeline] arguments for debug builds and may
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/debug.dart:138]: Can we improve the docs here a little bit and describe what additional information one can expect with these flags enabled and what the "other conditions" are? Right now, just from reading this doc I have no idea when I would ever turn this flag on...

File: packages/flutter/lib/src/rendering/debug.dart [hunk 3]
```diff
@@ -145,6 +145,22 @@ bool debugProfileLayoutsEnabled = false;
 ///    areas are being excessively repainted.
 bool debugProfilePaintsEnabled = false;
 
+/// Adds debugging information to the args of [Timeline] events related to
```
Comment (goderbauer) [inline packages/flutter/lib/src/rendering/debug.dart:148]: Same here for these flags.

File: packages/flutter/lib/src/widgets/debug.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/flutter/lib/src/widgets/debug.dart]: improved the docs. ptal.

File: packages/flutter/lib/src/widgets/debug.dart [hunk 4]
```diff
@@ -132,6 +132,15 @@ bool debugProfileBuildsEnabled = false;
 ///    for every widget and has a higher overhead cost.
 bool debugProfileBuildsEnabledUserWidgets = false;
 
+/// Adds debugging information to [Timeline] events related to [Widget] builds.
+///
+/// This flag will only add [Timeline] event arguments for debug builds.
+/// Additional arguments will be added for the "BUILD" [Timeline] event and for
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/debug.dart:138]: Let's also document what these arguments are, e.g. additional stats around widget dirtyness and for widgets you get the diagnostics information provided by the widget (i.e. the properties of the widget). The latter can make it easier to identify a certain widget in the timeline.

File: packages/flutter/lib/src/rendering/debug.dart [hunk 5]
```diff
@@ -145,6 +145,24 @@ bool debugProfileLayoutsEnabled = false;
 ///    areas are being excessively repainted.
 bool debugProfilePaintsEnabled = false;
 
+/// Adds debugging information to [Timeline] events related to [RenderObject]
+/// layouts.
+///
+/// This flag will only add [Timeline] event arguments for debug builds.
+/// Additional arguments will be added for the "LAYOUT" timeline event and for
```
Comment (goderbauer) [inline packages/flutter/lib/src/rendering/debug.dart:152]: same here: you get additional stats on dirtyness and the RenderObject events contain their diagnostic information.

File: packages/flutter/lib/src/rendering/debug.dart [hunk 6]
```diff
@@ -145,6 +145,24 @@ bool debugProfileLayoutsEnabled = false;
...
+/// all [RenderObject] layout [Timeline] events, which are the events that are
+/// added when [debugProfileLayoutsEnabled] is true.
+bool debugEnhanceLayoutTimelineArguments = false;
+
+/// Adds debugging information to [Timeline] events related to [RenderObject]
+/// paints.
+///
+/// This flag will only add [Timeline] event arguments for debug builds.
+/// Additional arguments will be added for the "PAINT" timeline event and for
```
Comment (goderbauer) [inline packages/flutter/lib/src/rendering/debug.dart:161]: and here.

File: packages/flutter/lib/src/rendering/debug.dart (hunk 5 above)
Comment (kenzieschmoll) [reply packages/flutter/lib/src/rendering/debug.dart:156]: done

File: packages/flutter/lib/src/rendering/debug.dart (hunk 6 above)
Comment (kenzieschmoll) [reply packages/flutter/lib/src/rendering/debug.dart:177]: done

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**